tests/scripts/all.sh

```diff
@@ -953,6 +953,17 @@ component_test_no_max_fragment_length_small_ssl_out_content_len () {
     if_build_succeeded tests/ssl-opt.sh -f "Max fragment length\|Large buffer"
 }
 
+component_test_when_no_ciphersuites_have_mac () {
+    msg "build: when no ciphersuites have MAC"
+    scripts/config.pl unset MBEDTLS_CIPHER_NULL_CIPHER
+    scripts/config.pl unset MBEDTLS_ARC4_C
+    scripts/config.pl unset MBEDTLS_CIPHER_MODE_CBC
+    make
+
+    msg "test: !MBEDTLS_SSL_SOME_MODES_USE_MAC"
+    make test
```

Comment:
I think this is an interesting test addition that would merit a run of `ssl-opt.sh`.

Reply:
On second thought, this is probably not necessary because `config-suite-b` already exercises a configuration disabling MAC modes entirely.

Reply:
I was wondering why this isn't caught by `config-suite-b` and `depends-cipher`. So `config-suite-b` doesn't enable `ENTROPY_THEN_MAC`, `depends-cipher` doesn't exist, and I think the current proposal for `depends-cipher` wouldn't catch this because it doesn't reach any AEAD-only configuration. I have a more general `depends.py` in the works that would catch this, but it's far from ready.